internal-enrichment/shodan-internetdb/src/shodan_internetdb/connector.py

```diff
@@ -73,166 +59,57 @@ def _process_message(self, data: Dict) -> str:
-            """Format a list of primitives into a Markdown list"""
-            return "".join(f"\n- {name}" for name in alist) or "n/a"
-
-        value = observable["value"]
-        abstract = f"Shodan InternetDB enrichment of {value}"
-        content = f"""```
-Shodan InternetDB:
-------------------
-Hostnames: {format_list(result.hostnames)}
-------------------
-Software: {format_list(result.cpes)}
-------------------
-Vulnerabilities: {format_list(result.vulns)}
-------------------
-Ports: {format_list(result.ports)}
-------------------
-```
-"""
-        note = stix2.Note(
-            id=Note.generate_id(datetime.now().isoformat(), content),
-            created_by_ref=self._identity_id,
-            object_marking_refs=[self._object_marking_id],
-            abstract=abstract,
-            content=content,
-            object_refs=[observable["id"]],
-            allow_custom=True,
+        return self._send_bundle(
+            stix_objects + self.converter.create_stix_objects(stix_observable, result)
         )
-        return note
-
-    def _process_domains(
-        self,
-        observable: Dict[str, Any],
-        result: ShodanResult,
-    ) -> list:
-        """
-        Add additional domains to the observable
-        :param observable: Observable data
-        :param result: Shodan data
-        :return: None
-        """
-        stix_objects = []
-        for name in result.hostnames:
-            log.debug("Adding domain %s", name)
-            stix_domain = stix2.DomainName(
-                value=name,
-                object_marking_refs=[self._object_marking_id],
-                resolves_to_refs=[observable["id"]],
-                custom_properties={"created_by_ref": self._identity_id},
-            )
-            stix_objects.append(stix_domain)
-        return stix_objects
-
-    def _process_tags(
-        self,
-        observable: Dict[str, Any],
-        result: ShodanResult,
-    ) -> Dict:
-        """
-        Add additional tags to the observable
-        :param observable: Observable data
-        :param result: Shodan data
-        :return: None
-        """
-        for name in result.tags:
-            OpenCTIStix2.put_attribute_in_extension(
-                observable, STIX_EXT_OCTI_SCO, "labels", name, True
-            )
-        return observable
 
-    def _process_vulns(
-        self,
-        observable: Dict[str, Any],
-        result: ShodanResult,
-    ) -> list:
+    def process_message(self, data: dict[str, Any]) -> str:
+        try:
+            return self._process_message(data)
+        except (
+            ShodanInternetDbInvalidIPv4AddressError,
+            ShodanInternetDbEntityNotInScopeError,
+            ShodanInternetDbApiError,
+            ShodanInternetDbNotFoundError,
+            ShodanInternetDbExtractAndCheckMarkingsError,
+            ShodanInternetDbInvalidTlpLevelError,
+        ) as e:
+            self.helper.connector_logger.error(e)
```

Comment:
Discussed with the product team:
```suggestion
            self.helper.connector_logger.warning(e)
```


New directives:

    Info: Used when we know the connector might not handle a certain data type and we assume it.

    Warning: Used when a packet to be processed is absent, corrupted, etc., and is skipped.

    Error: Used when the connector is broken due to a code error.